### PR TITLE
FIX on prometheus scrape url

### LINKED ISSUE #298: wrong port (`None`) in Prometheus scrape job
### Bug Description

When relationg to Traefik, Loki inform a wrong port: `None`

imagen

### To Reproduce

1. Deploy cos-lite `edge` (Loki Rev 96)
2. Check Prometheus targets

### Environment

- cos-lite edge

### Relevant log output

```shell
.
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Abuelodelanada**
## Issue

This PR fixes #298

imagen

## Solution

`scrape_jobs()` refactored by making use of the port only if it's defined

### CODE REVIEW

File: src/charm.py
Comment (sed-i) [inline src/charm.py:272]: Because of
```python
>>> urlparse("scheme://hostname").path
''
>>> urlparse("scheme://hostname/").path
'/'
```
we probably need:
```suggestion
        metrics_path = f"{external_url.path.rstrip("/")}/metrics"
```

**Comment by lucabello** [comment]
Unit tests are blocked by the `ops` issue that was mentioned, which should be fixed soon.
The changes LGTM, so go ahead and merge this once the CI is green! :)

**Review by lucabello — APPROVED** [review]

**Pull request merged**
